workers/unity/Packages/io.improbable.gdk.core/.codegen/Templates/CommandDiffStorageGenerator.tt

```diff
@@ -111,12 +110,9 @@ namespace <#= qualifiedNamespace #>
                 }
 
                 var responseIndex = responseStorage.GetResponseIndex(requestId);
-                if (responseIndex < 0)
-                {
-                    return ReceivedMessagesSpan<<#= receivedResponseType #>>.Empty();
-                }
-
-                return new ReceivedMessagesSpan<<#= receivedResponseType #>>(responseStorage, responseIndex, 1);
+                return !responseIndex.HasValue
```

Comment:
Nit - I'd invert this such that its `responseIndex.HasValue ? responseStorage.Slice(responseIndex.Value, 1) : MessagesSpan<BarCommand.ReceivedResponse>.Empty()`